arrow-fx-coroutines-stream/src/main/kotlin/arrow/fx/coroutines/stream/predef.kt

```diff
@@ -35,3 +35,8 @@ internal fun <A, B> Iterator<A>.flatMap(f: (A) -> Iterator<B>): Iterator<B> =
     override fun next(): B =
       (if (hasNext()) cur else empty).next()
   }
+
+/** Represents a unique identifier using object equality. */
+internal class Token {
+  override fun toString(): String = "Token(${Integer.toHexString(hashCode())})"
+}
```

Comment:
How unique are `hashCode()` calls to this class? I actually never tested that :sweat_smile:

Reply:
https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#hashCode()

> As much as is reasonably practical, the hashCode method defined by class Object does return distinct integers for distinct objects. (This is typically implemented by converting the internal address of the object into an integer, but this implementation technique is not required by the JavaTM programming language.)

So "As much as is reasonably practical" 😅